src/executor/evaluate/mod.rs

```diff
@@ -197,6 +199,72 @@ pub async fn evaluate<'a, T: 'static + Debug>(
         Expr::Wildcard | Expr::QualifiedWildcard(_) => {
             Err(EvaluateError::UnreachableWildcardExpr.into())
         }
+        Expr::Case {
+            operand,
+            when_then,
+            else_result,
+        } => {
+            let operand = match operand {
+                Some(expr) => Some(eval(expr).await?),
+                None => None,
+            };
+
+            let when_then: Vec<(Evaluated, Evaluated)> = stream::iter(when_then.iter())
+                .map(Ok::<_, Error>)
+                .and_then(|(when, then)| async move { Ok((eval(when).await?, eval(then).await?)) })
+                .try_collect::<Vec<_>>()
+                .await?;
+
+            let else_result = match else_result {
+                Some(result) => Some(eval(result).await?),
+                None => None,
+            };
+
+            let (whens, thens) = when_then
+                .clone()
+                .into_iter()
+                .unzip::<_, _, Vec<_>, Vec<_>>();
+
+            let results = match &else_result {
+                Some(er) => [thens.clone(), [er.to_owned()].to_vec()].concat(),
+                None => thens.clone(),
+            }
+            .into_iter()
+            .map(|result| result.try_into())
+            .collect::<Result<Vec<Value>>>()?;
```

Comment:
```suggestion
            let results = else_result
                .as_ref()
                .map(|er| [thens.clone(), vec![er.to_owned()]].concat())
                .unwrap_or_else(|| thens.clone())
                .into_iter()
                .map(TryInto::try_into)
                .collect::<Result<Vec<Value>>>()?;
```

Reply:
Lots of useful new things in here! thanks